File: Diaguard/src/main/java/com/faltenreich/diaguard/export/pdf/print/PdfText.java
package com.faltenreich.diaguard.export.pdf.print;

import com.pdfjet.Page;
import com.pdfjet.Paragraph;
import com.pdfjet.Text;

import java.util.List;

public class PdfText extends Text {

    private float height;

    public PdfText(List<Paragraph> paragraphs) throws Exception {
        super(paragraphs);
    }

    public float getHeight() {
        return height;
    }

    @Override
    public float[] drawOn(Page page, boolean draw) throws Exception {
        float[] points = super.drawOn(page, draw);
        float startY = getBeginParagraphPoints().get(0)[1];
        float endY = points[1];
        height = endY - startY;
        return points;
    }
}

File: feature/export/src/androidMain/kotlin/com/faltenreich/diaguard/export/pdf/print/PdfText.kt
package com.faltenreich.diaguard.export.pdf.print

import android.graphics.Paint
import android.graphics.PointF
import android.graphics.Rect
import android.util.Size

internal class PdfText(
    private val text: String,
    private val paint: Paint,
) : PdfDrawable {

    override fun getSize(page: PdfPage): Size {
        val bounds = Rect()
        paint.getTextBounds(text, 0, text.length, bounds)
        return Size(bounds.width(), bounds.height())
    }

    override fun drawOn(page: PdfPage, position: PointF) {
        page.canvas.drawText(text, position, paint)
    }
}